src/applications/Job.java

```diff
@@ -49,10 +49,37 @@ public void setArrivalTime(int arrivalTime) {
     public int getId() {
         return id;
     }
-
+    /**
+     * put the next job on the machine queue.
+     * @param machineShopSimulator
+     * @param p // the machine to get
+     */
 	void putJobOnMachineQueue(MachineShopSimulator machineShopSimulator, int p) {
         Machine machine = machineShopSimulator.getMachine(p);
         machine.getJobQ().put(this);
 	}
 
+	/**
+	 * move theJob to machine for its next task
+	 * @param machineShopSimulator
+	 * @param simulationResults
+     * @param eList
+	 * @return false if no next task, true if has next task.
+	 */
+	boolean moveToNextMachine(MachineShopSimulator machineShopSimulator, SimulationResults simulationResults, EventList eList) {
+	    if (getTaskQ().isEmpty()) {// the job has no next task; return false
```

Comment:
It seems weird to be calling a getter for an internal field; why not just look at it directly?